src/common/common.go

```diff
@@ -168,6 +168,15 @@ func BuildHostsMapIPAddressBased(inventoryHostsMap map[string]inventory_client.H
 	return knownIpAddresses
 }
 
+func LogIfHostIpWasChanged(log logrus.FieldLogger, node v1.Node, IPAddressMap map[string]inventory_client.HostData) {
+	for _, ip := range node.Status.Addresses {
+		_, exists := IPAddressMap[ip.Address]
+		if !exists && ip.Type == v1.NodeInternalIP {
```

Comment:
Seems that you should check the map only for the nodeInternalIP.
Also, consider passing to this function just the node addresses or just the nodeInternalIP

Reply:
Will require for loop outside the function to search for internal ip. Don't want to handle it outside actually.

Reply:
```suggestion
	for _, ip := range node.Status.Addresses {
		if ip.Type == v1.NodeInternalIP {
			_, exists := IPAddressMap[ip.Address]
			if !exists {
```